component/http/handler.go

```diff
@@ -62,44 +62,77 @@ func determineEncoding(r *http.Request) (string, encoding.DecodeFunc, encoding.E
 	var enc encoding.EncodeFunc
 	var dec encoding.DecodeFunc
 	var ct string
+	var err error
 
 	if cok {
-		switch cth[0] {
-		case "*/*", json.Type, json.TypeCharset:
-			enc = json.Encode
-			dec = json.Decode
-			ct = json.TypeCharset
-		case protobuf.Type, protobuf.TypeGoogle:
-			enc = protobuf.Encode
-			dec = protobuf.Decode
-			ct = protobuf.Type
-		default:
-			return "", nil, nil, errors.New("content type header not supported")
+		ah := getMultiValueHeaders(cth[0])
+		for _, v := range ah {
+			ct, dec, enc, err = getSingleHeaderEncoding(v)
```

Comment:
Question: Let's assume that we have 3 encodings and the 3rd is correct but the previous 2 are not supported, should we fail? Maybe we need just a correct one.